oximeter/db/schema/replicated/db-init-1.sql

```diff
@@ -0,0 +1,204 @@
+
+CREATE TABLE IF NOT EXISTS oximeter.measurements_f64_local ON CLUSTER oximeter_cluster
+(
+    timeseries_name String,
+    timeseries_key UInt64,
+    timestamp DateTime64(9, 'UTC'),
+    datum Nullable(Float64)
+)
+ENGINE = ReplicatedMergeTree('/clickhouse/tables/{shard}/measurements_f64_local', '{replica}')
+ORDER BY (timeseries_name, timeseries_key, timestamp)
+TTL toDateTime(timestamp) + INTERVAL 30 DAY;
+
+CREATE TABLE IF NOT EXISTS oximeter.measurements_f64 ON CLUSTER oximeter_cluster
+AS oximeter.measurements_f64_local
+ENGINE = Distributed('oximeter_cluster', 'oximeter', 'measurements_f64_local', xxHash64(splitByChar(':', timeseries_name)[1]));
+
+CREATE TABLE IF NOT EXISTS oximeter.measurements_cumulativef64_local ON CLUSTER oximeter_cluster
+(
+    timeseries_name String,
+    timeseries_key UInt64,
+    start_time DateTime64(9, 'UTC'),
+    timestamp DateTime64(9, 'UTC'),
+    datum Nullable(Float64)
+)
+ENGINE = ReplicatedMergeTree('/clickhouse/tables/{shard}/measurements_cumulativef64_local', '{replica}')
+ORDER BY (timeseries_name, timeseries_key, start_time, timestamp)
+TTL toDateTime(timestamp) + INTERVAL 30 DAY;
+
+CREATE TABLE IF NOT EXISTS oximeter.measurements_cumulativef64 ON CLUSTER oximeter_cluster
+AS oximeter.measurements_cumulativef64_local
+ENGINE = Distributed('oximeter_cluster', 'oximeter', 'measurements_cumulativef64_local', xxHash64(splitByChar(':', timeseries_name)[1]));
+
+/* The field tables store named dimensions of each timeseries.
+ *
+ * As with the measurement tables, there is one field table for each field data
+ * type. Fields are deduplicated by using the "replacing merge tree", though
+ * this behavior **must not** be relied upon for query correctness.
+ *
+ * The index for the fields differs from the measurements, however. Rows are
+ * sorted by timeseries name, then field name, field value, and finally
+ * timeseries key. This reflects the most common pattern for looking them up:
+ * by field name and possibly value, within a timeseries. The resulting keys are
+ * usually then used to look up measurements.
+ *
+ * NOTE: We may want to consider a secondary index on these tables, sorting by
+ * timeseries name and then key, since it would improve lookups where one
+ * already has the key. Realistically though, these tables are quite small and
+ * so performance benefits will be low in absolute terms.
+ */
+CREATE TABLE IF NOT EXISTS oximeter.fields_i64_local ON CLUSTER oximeter_cluster
+(
+    timeseries_name String,
+    timeseries_key UInt64,
+    field_name String,
+    field_value Int64
+)
+ENGINE = ReplicatedReplacingMergeTree('/clickhouse/tables/{shard}/fields_i64_local', '{replica}')
+ORDER BY (timeseries_name, field_name, field_value, timeseries_key);
+
+CREATE TABLE IF NOT EXISTS oximeter.fields_i64 ON CLUSTER oximeter_cluster
+AS oximeter.fields_i64_local
+ENGINE = Distributed('oximeter_cluster', 'oximeter', 'fields_i64_local', xxHash64(splitByChar(':', timeseries_name)[1]));
```

Comment:
The goal here was twofold: 
1. Consistency with the other tables
2. Enable querying from any active replica of data using the `distributed` table routing.

However, the more I think about this, the less sense it makes without sharding. My gut is telling me that for full replication of data, any query to a single node will return the data from that node if it responds since the data is there locally. And since the client doesn't contact any other node directly, if that node is down then no automated routing can take place. Unfortunately, the behavior of distributed fully replicated tables is not documented anywhere AFAICT. We may have to ask the clickhouse team for clarification, as I'm not sure how to test this.

This also has an impact on the comment I added about local testing above. If we always serve data from the contacted clickhouse server node, then using oxql to query the distributed nodes, or sql to query the `_local` nodes directly should behave the same.

Reply:
Sounds good!

> Unfortunately, the behavior of distributed fully replicated tables is not documented anywhere AFAICT. We may have to ask the clickhouse team for clarification, as I'm not sure how to test this.

Hm, yeah, I also don't have much information about this.